### PR TITLE
Add julia-conda-packaging

### PR DESCRIPTION
**Pull request by mkitti**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

@conda-forge/help-julia @isuruf 

This recipe contains a single activate script. The activate script
1. Scans `${CONDA_PREFIX}/share/julia/conda_clones` for git clones of packages.
2. Inspects the Project.toml in each directory.
3. If the UUID of the package is not within the current project at the correct version, the script ...
4. Makes sure there is a .git folder
5. Adds the package to a `Vector` of `Pkg.PackageSpec`
6. If the `Vector` of `Pkg.PackageSpec` is not empty, then it adds them all to the current project via single `Pkg.add`

Individual Julia packages thus should place a shallow git clone in `${CONDA_PREFIX}/share/julia/conda_clones`. Clones which have not been added to the current project will then be added.

### CODE REVIEW

**Comment by mkitti** [comment]
Why does it keep trying to test Windows?

**Comment by mkitti** [comment]
The next version of this involves moving the Julia code into it's own Julia package to take advantage of precompilation.

**Comment by mkitti** [comment]
@isuruf , could you take a look? Thanks.

File: recipes/julia-conda-packaging/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+# If your package is python based, we recommend using Grayskull to generate it instead:
+# https://github.com/conda-incubator/grayskull
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
```
Comment (isuruf) [inline recipes/julia-conda-packaging/meta.yaml:7]: Remove these

File: recipes/julia-conda-packaging/scripts/activate.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+clones_dir="${CONDA_PREFIX}/share/julia/conda_clones"
```
Comment (isuruf) [inline recipes/julia-conda-packaging/scripts/activate.sh:3]: Can you make this inline in line 7? We don't want to set variables that we don't want the user to see.

Comment (mkitti) [reply recipes/julia-conda-packaging/scripts/activate.sh]: Done in db247a0caddd5282b18af3331af4c3763d99f4c9

File: recipes/julia-conda-packaging/meta.yaml (hunk 1 above)
Comment (mkitti) [reply recipes/julia-conda-packaging/meta.yaml]: Done in db247a0caddd5282b18af3331af4c3763d99f4c9

**Comment by mkitti** [comment]
@isuruf, any other concerns?

**Comment by mkitti** [comment]
I hope everyone had a great Thanksgiving. I believe I handled everything from the last review. Could we take another look?

**Comment by mkitti** [comment]
That's understandable, and I'm being patient. It's just difficult to maintain focus on this effort as its dragged out over time.

Perhaps @SylvainCorlay would also be interested in looking this over as well.

**Pull request merged**